### PR TITLE
Changed authorization code to conform to requirements of Faraday 1.7.1

### PR DESCRIPTION
**Pull request by drstrangelooker**
Confirmed to remove the deprecation warning with Faraday 1.7.1 and later.

### CODE REVIEW

**Review by joeldodge79 — COMMENTED** [review]
gonna send you some instructions on vetting this with helltool first

**Comment by drstrangelooker** [comment]
I have a clean CI at https://github.com/looker/helltool/pull/65350.

File: looker-sdk.gemspec [hunk 1]
```diff
@@ -24,5 +24,5 @@ Gem::Specification.new do |s|
   s.require_paths = %w(lib)
   s.add_dependency 'jruby-openssl' if s.platform == :jruby
   s.add_dependency 'sawyer', '~> 0.8'
-  s.add_dependency 'faraday', ['>= 1.0', '< 2.0']
+  s.add_dependency 'faraday', ['>= 1.7.1', '< 2.0']
```
Comment (joeldodge79) [inline looker-sdk.gemspec:27]: I just realized this bump in minimal version - does pre-1.7.1 not work with the `http.headers[:authorization]` change?

Comment (drstrangelooker) [reply looker-sdk.gemspec]: I confirmed it works with 1.2.0 and later, so I set 1.2.0 as the min.

**Review by joeldodge79 — APPROVED** [review]
awesome, thanks!

**Pull request merged**